src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -429,64 +430,63 @@ public Location getUserLocation(String user)
+      Location otherLocation = locationEntityToLocationObj(entity);
+      if(otherLocation!=null) {
+        /*check distance between location to define if the are considered near*/
+        if(Math.abs(location.getLatitude()-otherLocation.getLatitude())<Datastore.LAT_LONG_DEGREE_DIFF 
+                && Math.abs(location.getLongitude()-otherLocation.getLongitude())<Datastore.LAT_LONG_DEGREE_DIFF)
         {
-          String idString = entity.getKey().getName();
-          UUID id = UUID.fromString(idString);
-          String text = (String) entity.getProperty("text");
-          String otherUser = (String) entity.getProperty("user");
-          Location locationNear = new Location(id, otherLatitude, otherLongitude, text, otherUser);
-          /*get user of the location to check if it is charity type before adding it to locations*/
-          User locationUser = getUser(otherUser);
-          if(locationUser!=null && locationUser.getType()!=null && locationUser.getType()==User.CHARITY_TYPE) {
-            locations.add(locationNear);
-          }
+          /*add the near loaction to list*/
+          locations.add(otherLocation);
         }
-      } catch (Exception e) {
-        System.err.println("Error reading location.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
       }
     }
     return locations;
   }
-
-  /*return list of charity locations near a user location*/
-  public List<Location> getAllNearLocations(Location location) {
-    String user = location.getUser();
-    List<Location> locations = new ArrayList<>();
-          Query allLocations =
-       new Query("Location")
-               .setFilter(new Query.FilterPredicate("user", FilterOperator.NOT_EQUAL, user));
-    PreparedQuery allLocationresults = datastore.prepare(allLocations);
-    for (Entity entity : allLocationresults.asIterable()) {
-      try {
-        double otherLongitude = Double.parseDouble(entity.getProperty("longitude").toString());
-        double otherLatitude = Double.parseDouble(entity.getProperty("latitude").toString());
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String text = (String) entity.getProperty("text");
-        String otherUser = (String) entity.getProperty("user");
-        Location locationNear = new Location(id, otherLatitude, otherLongitude, text, otherUser);
-        locations.add(locationNear);
-      } catch (Exception e) {
-        System.err.println("Error reading location.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
+  
+  /*returns a list of charity-user locations near a specific location*/
+  public List<Location> getAllNearCharityLocations(Location location) {   
+    List<Location> nearCharitylocations = new ArrayList<>();
+    List<Location> allNearLocations = getAllNearLocations(location);
+    
+    if(allNearLocations==null)
```

Comment:
If I understand correctly, getAllNearLocations never returns null, since it is assigned an empty arrayList on line 454.

Reply:
Yes, your right! I'll fix it! Thanks :)